README.md

```diff
@@ -1,2 +1,555 @@
+1
+```
+
+### Help page
+
+Each CLI in {Name} has a built-in help page, which is automatically generated.
+
+For example, for the following function:
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+@cli
+def entry(some: Arg):
+    print(some)
+
+if __name__ == '__main__':
+    cli()
+```
+
+It's possible to invoke the help page using the flag `--help`:
+```
+$ python test.py --help
+usage: entry [-h] some 
+
+positional arguments:
+  some
+
+optional arguments:
+  -h, --help  show this help message and exit
+```
+
+Note the lack of the overall program description as well as the description of the `some` argument.
+
+To override the program description, one could write a simple doc-comment for the CLI function:
+To assign a description to an argument, the `description` parameter of the argument constructor type should be used.
+
+Consider the following example as a demonstration of both possibilities:
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+@cli
+def entry(some: Arg('some argument')):
+    """A simple demonstration program."""
+    print(some)
+
+if __name__ == '__main__':
+    cli()
+```
+
+This produces the following help page:
+```
+$ python test.py --help
+usage: entry [-h] some 
+
+A simple demonstration program.
+
+positional arguments:
+  some        some argument
+
+optional arguments:
+  -h, --help  show this help message and exit
+```
+
+### Short name
+
+Usually, both flags and options come with a shortcut syntax.
+For example, instead of writing:
+```
+$ python test.py --some 1
+```
+
+One could write:
+```
+$ python test.py -s 1
+```
+
+To define a shortcut letter for a flag or an option, the `short` parameter of either `Flag` or `Opt` annotations should be used:
+```py
+# test.py
+
+from {pyname} import cli, Flag
+
+@cli
+def entry(some: Flag(short='s')):
+    print(some)
+
+if __name__ == '__main__':
+    cli()
+```
+
+This allows invoking the CLI with the following syntax:
+```
+$ python test.py -s
+True
+```
+
+### Prefix
+
+Flags and options are usually called with the `-` prefix (in short and long variations).
+To override this behavior, the `prefix` parameter of either `Flag` or `Opt` annotations should be used.
+
+Consider the following example:
+```py
+# test.py
+
+from {pyname} import cli, Flag
+
+@cli
+def entry(some: Flag(prefix='+')):
+    print(some)
+
+if __name__ == '__main__':
+    cli()
+```
+
+Then the command line interface could be invoked as follows:
+```
+$ python test.py ++some
+True
+```
+
+### Types
+
+By default, the value that comes from the CLI, if it's an `Opt` or an `Arg`, is of `str` type.
+
+Consider the following example:
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+@cli
+def entry(some: Arg):
+    print(type(some))
+
+if __name__ == '__main__':
+    cli()
+```
+
+If an integer value is passed to the command line, the following output is produced:
+```
+$ python test.py 1
+<class 'str'>
+```
+
+To enforce the value to be an `int`, the following syntax should be used:
```

Comment:
Maybe `convert` would look better here instead of `enforce`?

Reply:
Fixed!